src/main/java/org/interledger/btp/BilateralTransferProtocolError.java

```diff
@@ -0,0 +1,198 @@
+package org.interledger.btp;
```

Comment:
Wow, our Interface pattern has a _TON_ of boiler-plate! In the absence of moving this project to Kotlin (ha-ha), we should seriously consider using the [Immutables](http://immutables.github.io/) library. 

Basically, you just define an interface, and then during the build phase the library will generate immutable implementations of each class, complete with builders that are _very_ nice to deal with. Even better, the dependency for Immutables has a `provided` scope, (the library is used only for compilation), so it's not really a dependency that anyone using the library would need to worry about.

It saves _so_ much time, and I've used it successfully in many production projects with no issues.

Reply:
As an example, this would be the implemenation for `BilateralTransferProtocolFulfilll`:

```
@Value.Immutable
public interface BilateralTransferProtocolFulfill extends BilateralTransferProtocolTransferPacket {
  Fulfillment getFulfillment();
}
```

That's it - free immutable implementation, free builder, extensible, smart/configurable (and free) equals+hashcode+toString, compatible with Jackson, etc, etc...